NitroxLauncher/App.xaml.cs

```diff
@@ -30,6 +30,31 @@ protected override void OnStartup(StartupEventArgs e)
             {
                 MessageBox.Show("Nitrox launcher should not be executed from a temporary directory. Install Nitrox launcher properly by extracting ALL files and moving these to a dedicated location on your PC.", "Invalid working directory", MessageBoxButton.OK, MessageBoxImage.Error);
                 Environment.Exit(1);
+                Shutdown();
+            }
+
+            if (!RoleDetection.isAppRunningInAdmin()) {
+
+                if (Directory.GetCurrentDirectory().StartsWith(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), StringComparison.OrdinalIgnoreCase))
```

Comment:
You should be able to just get the subnautica directory from the launcherlogic instance
```suggestion
                if (LauncherLogic.Instance.SubnauticaPath.StartsWith(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), StringComparison.OrdinalIgnoreCase))
```

Reply:
He won't work, LauncherLogic isn't up yet while entering the application